block/local/adapter_test.go

```diff
@@ -141,3 +142,56 @@ func TestLocalCopy(t *testing.T) {
 		t.Errorf("expected to read \"%s\" as written, got \"%s\"", contents, string(got))
 	}
 }
+
+func TestRemove(t *testing.T) {
```

Comment:
This doesn't test what happens when an object is removed and *not* all directories are deleted.  E.g. add `a/b/c/d`, `/a/b/e/f` and remove `/a/b/e/f`.

Reply:
added such test